README.md

```diff
@@ -1,4 +1,116 @@
-# kernel-modules
+kernel-modules Cookbook
+=======================
 
-TODO: Enter the cookbook description here.
+[![Cookbook Version][cookbook_version]][cookbook]
+[![Build Status][build_status]][build_status]
 
+Configure and Load kernel modules.
+
+Requirements:
+-------------
+### Platforms
+* Centos 6.x/7.x
+* Rhel 6.x/7.x
+
+### Chef
+* Chef version: >= 12.5.1
+
+Attributes
+----------
+#### kernel_modules
+* `node['kernel_modules']['packages']` - used to make the packages requirements
+* `node['kernel_modules']['modules_load.d']` - used to define the `modules_load.d` directory
+* `node['kernel_modules']['modprobe.d']` - use to define the `modprobe.d` directory
+* `node['kernel_modules']['modules']` - use to define the list of modules you want to load/configure
+
+Usage
+-----
+Using this cookbook will load and configure every module in the attribute: `node['kernel_modules']['modules']`.
+
+For instance:
+``` ruby
+$ cat roles/modules.rb
+name 'Load modules'
+description 'Example for using kernel modules'
+
+run_list(
+  'kernel-modules'
+)
+
+default_attributes(
+kernel_modules:
+  modules: {
+    nfs: {
+      options: [
+        'enable_ino64=1',
+        'nfs4_disable_idmapping=0',
+      ],
+      alias: 'nfs4',
+      action: [:load],
+      onboot: true,
+    },
+  },
+)
+```
+Providers & Resources
+---------------------
+## kernel_module
+This provider allow to define a kernel module
+
+### Actions
+Action   | Description
+---------|---------------------------
+configure | At the module at boot time (if needed) and set the module modprobe command
+load | Load the module once it is configured
+unload| Unload the module
+remove| Remove the modprobe configuration file and remove the module at boot time
+
+### Property
+Name | Description | Default| Type
+-----|-------------|--------|-----
+`module_name`| module name| name |_String_
+`onboot`| load the module at boot time| true| _Boolean_
+`reload`| allow the module to be reloaded if module configuration changes| false| _Boolean_
+`force_reload`| allow the module to be unloaded even if the module is used[^1]. | false | _Boolean_
+`alias`| set the modprobe command _alias_| nil| _Array_/_String_
+`options`| set the modprobe command _options_| nil| _Array/String/NilClass_
+`install`| set the modprobe command _install_| nil| _String/NilClass_
+`install`| set the modprobe command _install_| nil| _String/NilClass_
```

Comment:
The `install` property is described twice.
Moreover I don't understand what you mean by the `command _install_`.